packages/react-components/src/components/FileDownloadCards.tsx

```diff
@@ -57,6 +77,23 @@ export interface FileMetadata {
   previewUrl?: string;
 }
 
+/**
+ * Meta Data containing information about the uploaded file.
+ * @beta
+ */
+export type FileMetadata =
+  | CustomFileMetadata
+  | /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */ TeamsInteropFileMetadata;
+
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+/**
+ * @beta
+ * @param fileMetadata
+ * @returns TeamsInteropFileMetadata
+ */
+export const isTeamsInteropFileMetadata = (fileMetadata: FileMetadata): fileMetadata is TeamsInteropFileMetadata => {
+  return (fileMetadata as TeamsInteropFileMetadata).id !== undefined;
```

Comment:
can we use the FileMetadataAttachmentType here instead of `id`, `id` seems like something other types could have in future

Reply:
Checking against the type now